core/src/main/java/org/elasticsearch/index/percolator/PercolatorFieldMapper.java

```diff
@@ -190,22 +188,18 @@ static Query toQuery(QueryShardContext context, boolean mapUnmappedFieldsAsStrin
         // as an analyzed string.
         context.setAllowUnmappedFields(false);
         context.setMapUnmappedFieldAsString(mapUnmappedFieldsAsString);
-        context.parseFieldMatcher(context.getIndexSettings().getParseFieldMatcher());
```

Comment:
The shard context parseFieldMatcher IMHO should always be backed by `indexSettings.getParseFieldMatcher()`